openedx/features/badging/tests/test_helpers.py

```diff
@@ -279,3 +275,17 @@ def test_filter_earned_badge_by_joined_team_course_with_no_team(self):
 
         self.assertIsNone(course_team)
         self.assertEqual(earned_badges, list())
+
+    @mock.patch('openedx.features.badging.helpers.publish_notification_to_user')
+    def test_send_user_badge_notification(self, publish_notification_to_user):
+        """
+        Test badge notification is sent to user
+        :param publish_notification_to_user: mock send notification through edx notifications
+        :return: None
+        """
+        # register badge notification type
+        badge_handlers.register_notification_types(None)
+        newly_earned_badge = BadgeFactory()
+
+        badge_helpers.send_user_badge_notification(self.user, 'dummy_badge_url', newly_earned_badge.name)
+        assert publish_notification_to_user.called_once
```

Comment:
Lets use `assert_called_once_with()` instead

Reply:
for that we need `NotificationMessage`.Should we create it through factory or creating a new object inside this test would be fine?

Reply:
done